src/lib/components/healthselectorv2/HealthSelector.component.js

```diff
@@ -0,0 +1,233 @@
+//@flow
+import React from 'react';
+import styled from 'styled-components';
+import Select, { Option } from '../selectv2/Selectv2.component';
+import Icon from '../icon/Icon.component';
+import * as defaultTheme from '../../style/theme';
+
+type optionType = 'all' | 'healthy' | 'warning' | 'critical' | 'unknown';
+type optionValue = {
+  icon?: React.Node,
+  label?: string,
+  shortLabel?: string,
+  value?: string,
+  disabled?: boolean,
+};
+
+type optionsConfiguration = { [optionType]: optionValue };
+
+type Props = {
+  optionsConfiguration?: optionsConfiguration,
+  label?: String,
+  onChange?: (value: string) => void,
+  defaultValue?: string,
+};
+
+const SingleValueWrapper = styled.div`
+  color: ${defaultTheme.white};
+  margin-left: ${defaultTheme.spacing.sp2}
+  margin-right: ${defaultTheme.spacing.sp2}
+  max-width: calc(100% - 8px);
+  position: absolute;
+  display: flex;
+  align-items: center;
+  height: 21px;
+  top: 0;
+`;
+
+const CSSCircle = styled.span`
+  background-color: ${(props) => defaultTheme.brand[props.variant]};
+  border-radius: 50%;
+  display: flex;
+  width: 16px;
+  height: 16px;
+  align-items: center;
+  justify-content: center;
+`;
+
+const ThreeCirclesWrapper = styled.span`
+  display: flex;
+  width: 34px;
+  justify-content: center;
+  ${CSSCircle}:first-child {
+    margin-right: -20%;
+  }
+  ${CSSCircle}:last-child {
+    margin-left: -20%;
+  }
+  height: 16px;
+`;
+
+const ValueIcon = styled.span`
+  margin: 0 ${defaultTheme.spacing.sp8};
+  height: 100%;
+  width: 100%;
+`;
+
+const OptionIcon = styled.span`
+  width: 35px;
+  margin-right: ${defaultTheme.spacing.sp4};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const OptionWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  justify-content: flex-start;
+  overflow: hidden;
+  line-height: 18px;
+`;
+
+const OptionLabel = styled.span`
+  display: flex;
+  align-items: bottom;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: block;
+}
+`;
+
+const SelectStyle = styled(Select)`
+  .sc-select__control {
+    background-color: ${defaultTheme.defaultTheme.darkRebrand.buttonSecondary};
+
+    &.sc-select__control--is-focused {
+      border: 2px dashed ${defaultTheme.defaultTheme.darkRebrand.selectedActive};
+      background-color: ${defaultTheme.defaultTheme.darkRebrand
+        .buttonSecondary};
+    }
+    .sc-select__value-container {
+      overflow: visible;
+    }
+
+    &.sc-select__control--menu-is-open {
+      .sc-select__indicator {
+        transform: rotate(180deg);
+      }
+    }
+  }
+`;
+
+const icons = {
+  all: (
+    <ThreeCirclesWrapper>
+      <CSSCircle variant="statusHealthy" />
+      <CSSCircle variant="statusWarning" />
+      <CSSCircle variant="statusCritical" />
+    </ThreeCirclesWrapper>
```

Comment:
nit: I would preferably use a svg here (In my mind declarative always win over imperative for these things) 
```suggestion
    <svg viewBox="0 0 200 100" xmlns="http://www.w3.org/2000/svg">
  <circle cx="50" cy="50" r="50" fill={theme.brand["statusHealthy"]}/>
  <circle cx="75" cy="50" r="50" fill={theme.brand["statusWarning"]}/>
  <circle cx="100" cy="50" r="50" fill={theme.brand["statusCritical"]}/>
</svg>
```
should do the trick I think but I haven't tested